src/main/java/com/redhat/devtools/intellij/tektoncd/actions/OpenEditorAction.java

```diff
@@ -16,87 +16,44 @@
         try {
-            if (element instanceof PipelineNode) {
-                content = tkncli.getPipelineYAML(namespace, element.getName());
-                kind = KIND_PIPELINES;
-            } else if (element instanceof ResourceNode) {
-                content = tkncli.getResourceYAML(namespace, element.getName());
-                kind = KIND_RESOURCES;
-            } else if (element instanceof TaskNode) {
-                content = tkncli.getTaskYAML(namespace, element.getName());
-                kind = KIND_TASKS;
-            } else if (element instanceof ClusterTaskNode) {
-                content = tkncli.getClusterTaskYAML(element.getName());
-                kind = KIND_CLUSTERTASKS;
-            } else if (element instanceof ConditionNode) {
-                content = tkncli.getConditionYAML(namespace, element.getName());
-                kind = KIND_CONDITIONS;
-            } else if (element instanceof TriggerTemplateNode) {
-                content = tkncli.getTriggerTemplateYAML(namespace, element.getName());
-                kind = KIND_TRIGGERTEMPLATES;
-            } else if (element instanceof TriggerBindingNode) {
-                content = tkncli.getTriggerBindingYAML(namespace, element.getName());
-                kind = KIND_TRIGGERBINDINGS;
-            } else if (element instanceof ClusterTriggerBindingNode) {
-                content = tkncli.getClusterTriggerBindingYAML(namespace, element.getName());
-                kind = KIND_CLUSTERTRIGGERBINDINGS;
-            } else if (element instanceof EventListenerNode) {
-                content = tkncli.getEventListenerYAML(namespace, element.getName());
-                kind = KIND_EVENTLISTENER;
-            }
-        }
-        catch (IOException e) {
+            yamlAndKind = TreeHelper.getYAMLAndKindFromNode(element);
+        } catch (IOException e) {
             UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Error"));
         }
 
-        if (!content.isEmpty()) {
+        if (yamlAndKind != null && !yamlAndKind.first.isEmpty()) {
             Project project = anActionEvent.getProject();
 
             Optional<FileEditor> editor = Arrays.stream(FileEditorManager.getInstance(project).getAllEditors()).
-                    filter(fileEditor -> fileEditor.getFile().getName().startsWith(namespace + "-" + element.getName()) &&
+                    filter(fileEditor -> fileEditor.getFile().getName().startsWith(namespace + "-" + element.getName() + ".yaml") &&
                             fileEditor.getFile().getExtension().equals("yaml")).findFirst();
             if (!editor.isPresent()) {
-                createAndOpenVirtualFile(project, namespace, namespace + "-" + element.getName() + ".yaml", content, kind, null);
+                VirtualDocumentHelper.createAndOpenVirtualFile(project, namespace, namespace + "-" + element.getName() + ".yaml", yamlAndKind.first, yamlAndKind.second);
```

Comment:
I would keep only one method of `createAndOpenVirtualFile`. Maybe we should delete the one inside TektonAction (as they are the same) and update the one inside VirtualDocumentHelper to make it accepts the extra parameters used to pass the targetnode.